### PR TITLE
Containerized test functions for faas-server

### PR DESCRIPTION
**Pull request by johnmuth81**
This creates the function containers in docker. Still to do in other PR's: extract into directories for use by KRUN.

### CODE REVIEW

File: kontain-faas/server/test_funcs/test_func_km.c [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+#define HC_snapshot_getdata 504
+#define HC_snapshot_putdata 503
+
+int
+main(int argc, char *argv[])
+{
+  char *msg = getenv("FUNC_MSG");
+  printf("%s\n", msg);
+  int msglen = strlen(msg);
+  syscall(HC_snapshot_putdata, msg, msglen);
```
Comment (sv641) [inline kontain-faas/server/test_funcs/test_func_km.c:21]: This is not the expected format of output file.

**Review by sv641 — COMMENTED** [review]
There is an  issue in  implementation of HC_snapshot_putdata
we are using open it should be replaced with km_internal_open.

**Comment by sv641** [comment]
both 
HC_snapshot_getdata and HC_snapshot_putdata are leaking fd's.

**Comment by johnmuth81** [comment]
> both
> HC_snapshot_getdata and HC_snapshot_putdata are leaking fd's.

Do you mean in KM?

**Review by sv641 — APPROVED** [review]

**Comment by sv641** [comment]
> > both
> > HC_snapshot_getdata and HC_snapshot_putdata are leaking fd's.
> 
> Do you mean in KM?

yes.

**Pull request merged**